pkg/arvo/app/btc-node-hook.hoon

```diff
@@ -0,0 +1,373 @@
+  `[%error (handle-error jon)]
+::
+++  handle-error
+  |=  =json
+  ?~  json  ['' '' '']
+  %.  json
+  =,  dejs:format
+  =-  (ou -)
+  :~  =-  ['id' (uf '' (cu - (mu so)))]
+      |*(a=(unit) ?~(a '' u.a))
+      :-  'error'
+      =-  (uf ['' ''] -)
+      =-  (cu |*(a=(unit) ?~(a ['' ''] u.a)) (mu (ou -)))
+      :~  ['code' (uf '' no)]
+          ['message' (uf '' so)]
+  ==  ==
+::
+++  http-response
+  |=  [=wire response=client-response:iris]
+  ?.  ?=(%finished -.response)
+    [~ state]
+  =*  status    status-code.response-header.response
+  =/  rpc-resp=response:rpc:jstd
+    %-  httr-to-rpc-response
+    %+  to-httr:iris
+      response-header.response
+    full-file.response
+  ?.  ?=([%result *] rpc-resp)
+    ~&  [%error +.rpc-resp]
+    [~ state]
+  %-  handle-btc-response
+    (parse-response:btc-rpc:lib rpc-resp)
+::
+++  handle-btc-response
+  |=  btc-resp=btc-node-hook-response
+  ^-  (quip card _state)
+  :_  state
+  ^-  (list card)
+  ?+    -.btc-resp
+      ::  By default we just print all RPC responses that are not
+      ::  considered here explicitly for proper format printing or
+      ::  for being passed on to the store app.
+      ::
+      ~&  btc-resp
+      ~
+  ::
+      :: %abandon-transaction
+      :: %abort-rescan
+      :: %add-multisig-address
+      :: %backup-wallet
+      :: %bump-fee
+      %create-wallet
+    ^-  (list card)
+    =/  btc-store-req=btc-node-store-action
+      :+  %add-wallet   name.btc-resp
+      ?:(=('' warning.btc-resp) ~ (some warning.btc-resp))
+    [(btc-node-store-poke /store btc-store-req)]~
+  ::
+      :: %dump-privkey
+      :: %dump-wallet
+      :: %encrypt-wallet
+      :: %get-addresses-by-label
+  ::
+      %get-address-info
+    ^-  (list card)
+    ~&  [%address-info +.btc-resp]
+    ~
+  ::
+      %get-balance
+    ^-  (list card)
+    :_  ~
+    :*  %pass  /  %arvo  %d  %flog
+        %text  "amount={(trip +.btc-resp)}"
+    ==
+  ::
+      :: %get-balance
+      :: %get-new-address
+      :: %get-raw-change-address
+      :: %get-received-by-address
+      :: %get-received-by-label
+      :: %get-transaction
+      :: %get-unconfirmed-balance
+  ::
+      %get-wallet-info
+    ^-  (list card)
+    =/  btc-store-req=btc-node-store-action
+      [%update-wallet wallet-name.btc-resp +>:btc-resp]
+    [(btc-node-store-poke /update btc-store-req)]~
+  ::
+      :: %import-address
+      :: %import-multi
+      :: %import-privkey
+      :: %import-pruned-funds
+      :: %import-pubkey
+      :: %import-wallet
+      :: %key-pool-refill
+      :: %list-address-groupings
+      :: %list-labels
+      :: %list-lock-unspent
+      :: %list-received-by-address
+      :: %list-received-by-label
+      :: %lists-in-ceblock
+  ::
+      %list-transactions
+    ^-  (list card)
+    ~&  [%transactions +.btc-resp]
+    ~
+  ::
+      :: %list-unspent
+      :: %list-wallet-dir
+  ::
+      %list-wallets
+    ^-  (list card)
+    [(btc-node-store-poke /list [%list-wallets ~])]~
```

Comment:
So, a node response is telling us to... print the wallets to the screen? Why?

Reply:
This should ask the full node to list the wallets that it contains. The problems comes when we can have inconsistencies with the wallet the %store app has. Now, I agree that this kinda doesn't make much sense and will need to be addressed when deciding what's really the purpose of the store, and with the sync behavior in place. Fo this first pass I just assumed that the control is made only from you urbit (and to show the connection with the %store app) so the consistency is maintained, but of course, this breaks without the syncing feature. I can also ignore what's in the %store and print the response from the full node.